modules/LocationUtils.js

```diff
@@ -68,6 +68,17 @@ export const createLocation = (path, state, key, currentLocation) => {
   return location;
 };
 
+export const shouldReplace = (location, newPath, newState) => {
+  const nextLocation = createLocation(newPath, newState, null, location);
+
+  return (
+    location.pathname === nextLocation.pathname &&
+    location.search === nextLocation.search &&
+    location.hash === nextLocation.hash &&
+    valueEqual(location.state, nextLocation.state)
```

Comment:
`state` shouldn't be compared; replicating how an anchor work only depends on the "visible" location properties.

Reply:
Fixed